### PR TITLE
Initialize delivery_count header at 0 instead of None.

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Add changelog note detailing this.

### CODE REVIEW

**Comment by annatisch** [comment]
I'm not convinced this fix should be in uamqp. By setting it to 0 here, we imply that the service actually gave us a value, when in fact it was omitted.
I would prefer this fix to be in the SB SDK, where we know the exact intentions of the service concerning delivery count.

**Comment by annatisch** [comment]
Ignore my earlier comment. AMQP protocol specification determines that 0 is a default value for the delivery count.

**Comment by yunhaoling** [comment]
as per discussion OOB, tweak the test to accommodate encoding size change brought by delivery_count default value "None" -> "0"

File: HISTORY.rst [hunk 1]
```diff
@@ -3,13 +3,18 @@
 Release History
 ===============
 
+1.2.8 (2020-05-18)
```
Comment (yunhaoling) [inline HISTORY.rst:6]: 1.2.8 (unreleased)

Comment (yunhaoling) [inline HISTORY.rst:6]:
```suggestion
1.2.8 (unreleased)
```

Comment (yunhaoling) [reply HISTORY.rst:6]: (we can update the release date before we decide to release to not constrain our plan 😀)

**Pull request merged**